src/base/bittorrent/session.cpp

```diff
@@ -4905,10 +4907,10 @@ void Session::handleSessionStatsAlert(const lt::session_stats_alert *p)
 
     emit statsUpdated();
 
-    if (m_refreshEnqueued)
-        m_refreshEnqueued = false;
-    else
-        enqueueRefresh();
+    QTimer::singleShot(STATS_UPDATE_INTERVAL, this, [this]()
```

Comment:
Why not use `refreshInterval()` for it too?

Reply:
`refreshInterval` can be changed by the user and its default value is 1500ms. Speedplotview assumes speed is updated per second but actually before this PR, stats are updated on `refreshInterval` this caused lots of duplicate data in the calculation for the graph depending on the value of refreshInterval. So by updating stats per second, independent of `refreshInterval` I aims to solve this problem.

Previously session stats are also used to display global rates, but because of limitation in libtorrent they were not in-sync with the aggregate values of torrent speeds from transfer list (https://github.com/qbittorrent/qBittorrent/issues/12459) so with second commit we now calculate global download and upload rates by manually totalling individual torrent speed values. 

With this session stats are now only used in graphs.

Reply:
>`refreshInterval` can be changed by the user and its default value is 1500ms. Speedplotview assumes speed is updated per second

So you don't think it makes sense to make Speedplotview dependent on the actual refresh interval (instead of some hardcoded value) to have it configurable?

Reply:
> So you don't think it makes sense to make Speedplotview dependent on the actual refresh interval (instead of some hardcoded value) to have it configurable?

It simplifies code and reduce efforts plus session stats are now only used in graphs.

Reply:
>It simplifies code and reduce efforts plus session stats are now only used in graphs.

Personally, I've never even changed "refresh interval". But apparently it might make sense for someone to make the refresh less frequent? So, doesn't it make the same sense to make all refresh stuff less frequent for the same reason?

Reply:
> session stats are now only used in graphs.

Only there? Really?
Then couldn't it make sense to make it available on demand, instead of constantly keep it refreshed at the app core level?

Reply:
> Personally, I've never even changed "refresh interval". But apparently it might make sense for someone to make the refresh less frequent? So, doesn't it make the same sense to make all refresh stuff less frequent for the same reason?

It seems a bit difficult to make SpeedPlotView refresh Interval aware.